framework/src/actions/CreateDisplacedProblemAction.C

```diff
@@ -26,23 +26,35 @@ InputParameters validParams<InitDisplacedProblemAction>()
   return params;
 }
 
-InitDisplacedProblemAction::InitDisplacedProblemAction(InputParameters parameters) :
+CreateDisplacedProblemAction::CreateDisplacedProblemAction(InputParameters parameters) :
     Action(parameters)
 {
 }
 
-InitDisplacedProblemAction::~InitDisplacedProblemAction()
+CreateDisplacedProblemAction::~CreateDisplacedProblemAction()
 {
 }
 
 void
-InitDisplacedProblemAction::act()
+CreateDisplacedProblemAction::act()
 {
-  if (isParamValid("displacements"))
+  if (isParamValid("displacements") && _displaced_mesh.get() != NULL)
```

Comment:
I think it may be better to just:

```
if (_displaced_mesh.get() == NULL)
  mooseError("displacements were set but a displaced mesh wasn't created!")
```

Inside the `if` here...